sqlglot/dialects/snowflake.py

```diff
@@ -167,14 +167,21 @@ def _datatype_sql(self: Snowflake.Generator, expression: exp.DataType) -> str:
     return self.datatype_sql(expression)
 
 
-def _regexpilike_sql(self: Snowflake.Generator, expression: exp.RegexpILike) -> str:
-    flag = expression.text("flag")
```

Comment:
We can't use `text("flag")` anymore because `REGEXP_LIKE` in Snowflake expects a third argument (`parameters`) which doesn't have to be a string literal and so this call would return an empty string if, say, we got a column instead.